libs/gl-client/src/credentials.rs

```diff
@@ -211,12 +230,34 @@ impl Device {
     pub fn to_bytes(&self) -> Vec<u8> {
         self.to_owned().into()
     }
+
+    pub fn node_id(&self) -> anyhow::Result<Vec<u8>> {
+        get_node_id_from_tls_config(&self.tls_config())
+    }
 }
 
 impl TlsConfigProvider for Device {
     fn tls_config(&self) -> TlsConfig {
         tls::TlsConfig::with(&self.cert, &self.key, &self.ca)
     }
+
+    /// Returns a new Device instance with a custom identity.
+    /// Tries to load the CA from file and defaults to CA_RAW.
+    /// Rune is initialized to the default value.
+    fn with_identity<V>(cert: V, key: V) -> Self
+    where
+        V: Into<Vec<u8>>,
+    {
+        let ca =
+            load_file_or_default("GL_CA_CRT", CA_RAW).expect("Could not load file from GL_CA_CRT");
+
+        Self {
+            cert: cert.into(),
+            key: key.into(),
+            ca,
+            ..Default::default()
+        }
+    }
 }
```

Comment:
Do we really need a way for `Device` credentials to be constructed from a `cert` and a `key`? Developers should load the credentials from bytes or a path as they already contain the CA and the credentials. This kind of sidesteps the idea to make this simpler and goes one step back towards the original tls_config way.

Reply:
This is mostly for developers who onboarded with TlsConfigs and need to upgrade their credentials.